### PR TITLE
Add FXIOS-6507 [v122] show of QRCodeViewController from QRCodeCoordinator to production code 

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
### Add QRCodeCoordinator to production code.
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6507)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14601)

### Enable QRCodeCoordinatorRefactor feature by default.
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7445)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16490)

## :bulb: Description

Add `QRCodeCoordinator` to production code. Add `QRCodeDismissHandler` protocol to handle dismiss of `QRCodeViewController` from `QRCodeCoordinator`. Turn on by default QRCodeCoordinatorRefactor feature.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by lmarceau** [comment]
Thank you I'll have a look 🙏

**Comment by lmarceau** [comment]
#### First point
So I confirm that the QRCode isn't presented from the IntroController. Are you able to have a look into this issue to fix it? The error is the following:
```
Attempt to present <Client.QRCodeNavigationController: 0x10a024000> on <UINavigationController: 0x109838800> (from <UINavigationController: 0x109838800>) whose view is not in the window hierarchy.
```
So the navigation controller that is created in `QRCodeCoordinator` isn't on screen in that case, hence why it cannot present the `QRCodeNavigationController`

#### Second point
As for the `didFinish`, it's indeed currently not called (when I tested from the `urlBarDidPressQRButton` and deeplinking methods which are able to present the QR code view). There's a couple of `dismiss` calls in `QRCodeCoordinator` which I think instead of calling them we should instead delegate to the QR code coordinator to properly handle the dismissal (so the qr code coordinator gets then deinit)

Let me know if that makes sense!

**Comment by FilippoZazzeroni** [comment]
@lmarceau thank you for your tests. I've corrected everything and updated the unit tests. Can you test it again on a real device ?

File: nimbus-features/qrCodeCoordinatorRefactor.yaml [hunk 1]
```diff
@@ -15,5 +15,5 @@ features:
           enabled: false
       - channel: developer
         value:
-          enabled: false
```
Comment (lmarceau) [inline nimbus-features/qrCodeCoordinatorRefactor.yaml:18]: Could you enable by default and for beta as well? So the task https://github.com/mozilla-mobile/firefox-ios/issues/16490 is done too

**Review by lmarceau — COMMENTED** [review]
Great work, all cases mentioned are working. One other case that is not handled is if we dismiss the scan QR code by swipping the view away - then the coordinator dismiss call is not called (so not pressing the back button, this case is working).

I think to fix it you just need to put back:
```
        router.present(navigationController, animated: true) { [weak self] in
            guard let self = self else { return }
            self.parentCoordinator?.didFinish(from: self)
        }
```
When you present the `QRCodeNavigationController`

**Comment by FilippoZazzeroni** [comment]
Thank you @lmarceau. I've corrected everything.

**Review by lmarceau — APPROVED** [review]
🎉

**Pull request merged**